crates/walrus-sui/src/client/retry_client/failover.rs

```diff
@@ -108,29 +112,36 @@ impl<ClientT, BuilderT: LazyClientBuilder<ClientT> + std::fmt::Debug>
         assert!(lazy_client_builders.len() >= max_tries);
 
         Ok(Self {
-            max_tries,
-            state: Arc::new(Mutex::new(FailoverState {
-                client: lazy_client_builders[0].lazy_build_client().await?,
-                rpc_url: lazy_client_builders[0].get_rpc_url().map(|s| s.to_string()),
-                current_index: 0,
-            })),
             lazy_client_builders,
+            state: Arc::new(Mutex::new(None)),
```

Comment:
Just noting that this delays the call to `ensure_experimental_rest_endpoint_exists` that was otherwise happening eagerly. This might not be a problem, but just wanted to be sure it's been considered.

Reply:
Yes. I'm aware that and my personal opinion is that we don't really need that `ensure_experimental_rest_endpoint_exists`. Now, even if that fails, we just switch to the next client, which is also better than just failing the node directly.

Reply:
I agree. Maybe we should put a TODO in to remove ensure_experimental_rest...